python/services/dataservice/dmod/dataservice/service.py

```diff
@@ -489,6 +495,91 @@ async def _async_process_query(self, message: DatasetManagementMessage) -> Datas
         """
         return self._process_query(message)
 
+    def _build_forcing_config_for_realization(self, request: AbstractNgenRequest) -> Forcing:
+        """
+        Build a ::class:`Forcing` config object from to satisfy requirements of this request.
+
+        Function builds a ::class:`Forcing` config object as a part of the steps to create a ngen realization config
+        for the given request.  It is typically expected that the provided request will include a partial realization
+        config object that includes certain details.
+
+        Parameters
+        ----------
+        request: AbstractNgenRequest
+            An AbstractNgenRequest request that needs a realization config generated, and as part of that, a forcing
+            config.
+
+        Returns
+        -------
+        Forcing
+            Forcing config object to be used in building a ngen realization config to satisfy this request.
+        """
+        forcing_cfg_params = dict()
+
+        # Get the correct forcing dataset from associated requirement
+        # TODO: double check that this is being added when we do data checks
+        forcing_req = [r for r in request.data_requirements if r.category == DataCategory.FORCING][0]
```

Comment:
Ideally if that is an invariant that we are capturing in the concept of an `AbstractNgenRequest` we would enforce that there, so I guess I am more or less asking if that is an invariant? Or is it acceptable to have more than one forcing requirement?

Reply:
This is not a huge deal in the slightest, so if this is just a can of worms that im opening, please say so!

Reply:
_Probably_ ... but I'm not sure.  My biggest concerns at the moment related to multi-level hydrofabrics.  Will it make sense to keep distinct forcing datasets for individual layers?  Even if we end up still using a unified dataset, will we want/need to keep requirements separate per layer as we assemble those for a job request?  

I'd rather kick this can down the road for now.

Reply:
> Even if we end up still using a unified dataset, will we want/need to keep requirements separate per layer as we assemble those for a job request?

I'd not thought about it in that way. For practical reasons, more or less for evaluation purposes, I'd assumed forcing products would be pre-processed into a single dataset, but that may be a poor assumption.